lib/elixir/test/elixir/string_test.exs

```diff
@@ -185,6 +189,10 @@ defmodule StringTest do
     assert String.downcase("OLÁ", :ascii) == "olÁ"
   end
 
+  test "downcase/1 with turkish" do
+    assert String.downcase("Iİ", :turkish) == "ıi"
```

Comment:
The unicode standard defines two tests here ([scroll to the bottom of this page](http://www.unicode.org/Public/UCD/latest/ucd/SpecialCasing.txt)).  I am copying here for convenience:

```elixir
# Turkish and Azeri

# I and i-dotless; I-dot and i are case pairs in Turkish and Azeri
# The following rules handle those cases.

0130; 0069; 0130; 0130; tr; # LATIN CAPITAL LETTER I WITH DOT ABOVE
0130; 0069; 0130; 0130; az; # LATIN CAPITAL LETTER I WITH DOT ABOVE

# When lowercasing, remove dot_above in the sequence I + dot_above, which will turn into i.
# This matches the behavior of the canonically equivalent I-dot_above

0307; ; 0307; 0307; tr After_I; # COMBINING DOT ABOVE
0307; ; 0307; 0307; az After_I; # COMBINING DOT ABOVE

# When lowercasing, unless an I is before a dot_above, it turns into a dotless i.

0049; 0131; 0049; 0049; tr Not_Before_Dot; # LATIN CAPITAL LETTER I
0049; 0131; 0049; 0049; az Not_Before_Dot; # LATIN CAPITAL LETTER I

# When uppercasing, i turns into a dotted capital I

0069; 0069; 0130; 0130; tr; # LATIN SMALL LETTER I
0069; 0069; 0130; 0130; az; # LATIN SMALL LETTER I
```

Can you please make sure we have both lowercase tests?

Reply:
Does it mean that the actual SpecialCasing.txt is truncated?
I don't see the `#EOF` flag..
I'll add this one too, if you don't mind...

Reply:
It is truncated because the bottom of it needs to be implemented by hand, we cannot do it ourselves. Generally speaking, we don't touch the SpecialCasing.txt unless when updating Unicode.

Reply:
I noticed that it's the conditional mappings that are truncated..
So the one we should take care manually..
I'll try to implement that later this day..
Thanks for the hints..

Reply:
I'm happy to see Turkic language support for casing making a comeback - it gladdens my Unicode heart!  That leaves only Lithuanian in `SpecialCasing.txt` not supported after this PR.

Reply:
Hello @josevalim 
I revert back the SpecialCasing.txt

And I added support for downcasing in both case (with and without the combining dot above character).
As how I understand, I left the canonical equivalence of the dot for the default.

@kipcole9 
I'll be happy to add those little Lithuanian use cases..
Just give me a little time to dig into the details of how the accents are handled according to http://www.unicode.org/Public/UCD/latest/ucd/SpecialCasing.txt
In a couple of days it should be done..

Reply:
Do I need to merge the commits, or make a new PR?
I'm not used to contribute in public repos (but I'll be more than happy to do it more often!)

Reply:
@josevalim I did squash the commits